Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -288,6 +288,10 @@ void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher*, IPC::Connection
     if (!connectionIdentifier)
         return;
 
+#if PLATFORM(MAC) && USE(RUNNINGBOARD)
+    m_lifetimeAssertion = ProcessAssertion::create(xpc_connection_get_pid(connectionIdentifier.xpcConnection.get()), "Lifetime assertion"_s, ProcessAssertionType::Foreground);
```

Comment:
Isn't a background assertion enough here? Presumably, if the tab is foreground, the process throttler will take its own foreground assertion and this is just the backup assertion to make sure we never suspend on macOS?

Reply:
Good question. I'm not enabling the process throttler yet with this change. Instead I'm taking a foreground assertion on each process for the entirety of its process. This is nearly equivalent to what we do currently with leaking an os_transaction/boost. The reason is that I want to preserve existing behavior as much as possible with this change (note: this preserves App Nap).

Reply:
Ok. Sounds good.